src/main/java/org/jenkinsci/plugins/configfiles/buildwrapper/ManagedFile.java

```diff
@@ -110,16 +109,23 @@ public String getDisplayName() {
             return "";
         }
 
-        public ListBoxModel doFillFileIdItems(@AncestorInPath ItemGroup context, @AncestorInPath Item project, @AncestorInPath AccessControlled ac) {
+        public ListBoxModel doFillFileIdItems(@AncestorInPath ItemGroup context, @AncestorInPath Item project, @AncestorInPath AccessControlled ac,
+                                              @QueryParameter String fileId) {
             // You should have permission to configure your project in order to get the available managed files
+            ListBoxModel items = new ListBoxModel();
+            items.add("please select", "");
             if (project != null) {
-                project.checkPermission(Item.CONFIGURE);
+                if (!project.hasPermission(Item.EXTENDED_READ)) {
+                    items.add(fileId);
+                    return items;
+                }
             } else {
-                ac.checkPermission(Item.CREATE); // This to get the first parent to validate the authorization (Folder, View or Jenkins)
```

Comment:
Not sure why `Item.CREATE` was used here. The only way we can get here is when accessing `pipeline-syntax` at root. Configure seems more logical to me. E.g. the create permission doesn't allow to access the pipeline snippetizer on a job, it requires Configure or Extended_read

Reply:
Because you can have permission to create an item on Jenkins but not to configure on Jenkins level.